[expo-dev-menu] Add auth & profile screen

lukmccall opened the pull request:
# Why

Adds auth & profile screen to dev-menu

# How

- Reimplemented web browser functionalities in dev-menu (mostly copied the code from `expo-web-browser`)
- Implemented UI - similar to one in the home app
- Created simple local storage to store a session.
- Added the GraphQL client

# Test Plan

- bare-expo ✅   -  https://exponent-internal.slack.com/archives/C3XK84RRR/p1612967628256400

lukmccall committed:
[expo-dev-menu] Add auth & profile screen

lukmccall committed:
Fix CI

lukmccall committed:
Fix `isAuthenticated`

tsapeta commented on packages/expo-dev-menu/app/DevMenuWebBrowser.ts:
```diff
@@ -0,0 +1,91 @@
+let _onWebBrowserCloseAndroid: null | (() => void) = null;
+let _isAppStateAvailable: boolean = AppState.currentState !== null;
+
+function _onAppStateChangeAndroid(state: any) {
```
I dislike the idea of prefixing private functions with `_` 😅 If it's not exported then it's private 🤷‍♂️ Anyway, let's do as you think is best 😉

tsapeta commented on packages/expo-dev-menu/app/api/ApolloClient.ts:
```diff
@@ -0,0 +1,38 @@
+  uri: `${Endpoints.api.origin}/--/graphql`,
+
+  async request(operation): Promise<void> {
+    // TODO(lukmccall): Check if the connection is available
+    // const isConnected = await Connectivity.isAvailableAsync();
+    // if (!isConnected) {
+    //   throw new Error('No connection available');
+    // }
```
Did you forget to do/uncomment this? 👀

lukmccall committed:
[expo-dev-menu] Make dev-menu available in dev-launcher (#11944)

# Why

Continuation of https://github.com/expo/expo/pull/11915.

# How

- Detected if the dev-menu pacackage is available
- Added a way to open dev-menu on a `Profile` or `Settings` screen
- Added a UI that opens dev-menu when user interacts with dev-launcher

# Test Plan

- bare-expo

lukmccall committed:
Apply requested changes
